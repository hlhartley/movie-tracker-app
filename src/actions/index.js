export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
});

export const loginUser = (id, name) => ({
  type: 'LOGIN_USER',
  id,
  name
});

export const updateError = (message) => ({
  type: 'UPDATE_ERROR',
  message
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const showPopup = (bool) => ({
  type: 'SHOW_POPUP',
  showPopup: bool
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
});
