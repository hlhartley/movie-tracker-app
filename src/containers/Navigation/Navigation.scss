.user-nav,
.guest-nav {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: black;
    width: 100vw;
    height: 50px;
}

a {
    padding: 0 20px;
    color: $gold;
    text-decoration: none;
}

.logout-btn {
    border-radius: 10px;
    width: 90px;
    border: 1px solid black;
    background-color: $gold;
    font-weight: 700;
    padding: 5px;
    margin-right: 20px;
    &:hover {
        background-color: $darksilver;
    }
}

.fa-crown,
.fa-home {
    display: inline;
    font-size: 16px;
}