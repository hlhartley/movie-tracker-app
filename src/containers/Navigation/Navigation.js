import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { logoutUser, resetFavorites, toggleFavorite } from '../../actions';
import PropTypes from 'prop-types';

export class Navigation extends Component {
    constructor() {
        super()
    }

    handleLogout = () => {
        let { favoriteMovies, toggleFavorite, resetFavorites, logoutUser } = this.props
        if (favoriteMovies.length > 0) {
            favoriteMovies.forEach((favorite) => {
                toggleFavorite(favorite);
            })
            resetFavorites();
        }
        logoutUser();
    }

    render() {
        let { currentUser } = this.props
        if (currentUser) {
            return( 
            <nav className='user-nav'>
                <div className='left-side-nav'>
                    <NavLink to='/' className='nav'><i className="fas fa-home"></i> Home </NavLink>
                    <NavLink to='/favorites' className='nav'><i className='fas fa-crown'></i> Favorites </NavLink>
                </div>
                <div>
                    <button onClick={this.handleLogout} className='logout-btn'>Log out</button>
                </div>
            </nav>
            )
        } else {
            return(
                <nav className='guest-nav'>
                    <div>
                        <NavLink to='/' className='nav'><i className="fas fa-home"></i> Home </NavLink>
                    </div>
                    <div>
                        <NavLink to='/login' className='nav'>Login</NavLink> 
                        <NavLink to='/create-account' className='nav'>Create Account</NavLink> 
                    </div>
                </nav>
            )
        }
    }
}

export const mapStateToProps = (state) => ({
    currentUser: state.currentUser,
    favoriteMovies: state.favoriteMovies,
});

export const mapDispatchToProps = (dispatch) => ({
    logoutUser: () => dispatch(logoutUser()),
    toggleFavorite: (id) => dispatch(toggleFavorite(id)),
    resetFavorites: () => dispatch(resetFavorites()) 
});


Navigation.propTypes = {
    favoriteMovies: PropTypes.array,
    logoutUser: PropTypes.func,
    toggleFavorites: PropTypes.func,
    resetFavorites: PropTypes.func, 
}

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);

