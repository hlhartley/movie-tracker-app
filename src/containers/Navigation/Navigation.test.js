import React from 'react';
import { Navigation, mapStateToProps, mapDispatchToProps } from './Navigation';
import { shallow } from 'enzyme';
import { logoutUser, toggleFavorite, resetFavorites } from '../../actions';

describe('Navigation', () => {

    describe('Navigation component', () => {
        let wrapper;
        const mockLogoutUser = jest.fn();
        const mockToggleFavorite = jest.fn();
        const mockResetFavorites = jest.fn();
        beforeEach(() => {
            wrapper = shallow(<Navigation loginUser={mockLogoutUser} toggleFavorite={mockToggleFavorite} resetFavorites={mockResetFavorites}/>)
        })

        it('should match the snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        })
    });

    describe('mapStateToProps', () => {
        it('should return an object with a currentUser and favoriteMovies', () => {
            const mockState = {
                currentUser: null,
                favoriteMovies: [],
                errorStatus: '',
            }
            const expected = {
                currentUser: null,
                favoriteMovies: [],
            }

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);
        });
    });
    
    describe('mapDispatchToProps', () => {
        it('should call dispatch with a logoutUser action when logoutUser is called', () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = logoutUser();
            const mappedProps = mapDispatchToProps(mockDispatch);
            mappedProps.logoutUser();
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });

        it('should call dispatch with a toggleFavorite action when toggleFavorite is called', () => {
            const mockDispatch = jest.fn();
            const id =1;
            const actionToDispatch = toggleFavorite(id);
            const mappedProps = mapDispatchToProps(mockDispatch);
            mappedProps.toggleFavorite(id);
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });

        it('should call dispatch with a resetFavorites action when resetFavorites is called', () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = resetFavorites();
            const mappedProps = mapDispatchToProps(mockDispatch);
            mappedProps.resetFavorites();
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });
    });
})