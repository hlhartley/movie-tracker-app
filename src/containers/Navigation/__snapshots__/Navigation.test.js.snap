// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Navigation component should match the snapshot 1`] = `
<nav>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/login"
  >
    Login
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav"
    to="/create-account"
  >
    Create Account
  </NavLink>
</nav>
`;
